packages/clerk-js/src/utils/url.ts

```diff
@@ -231,6 +231,21 @@ export const appendAsQueryParams = (
   return base + (params.toString() ? '#/?' + params.toString() : '');
 };
 
+export const appendAsQueryParams = (
+  baseUrl: string | URL,
+  values: Record<string, string | null | undefined> = {},
+): string => {
+  const base = toURL(baseUrl);
+  for (const [key, val] of Object.entries(values)) {
+    if (!val) {
+      continue;
+    }
+    base.searchParams.append(camelToSnake(key), val);
+  }
+
+  return baseUrl.toString() + base.search;
```

Comment:
🔧 Since this is URL object, why do you have to concatenate its parts? I'd expect the function to return `base.href` or `base.toString()`.

Reply:
I though it's better than we don't add an origin to the url if a relative one was provided.

Reply:
So baseUrl can be just a path?

Reply:
Yes, it can be a string. Inside `toUrl()` we construct a url with `window.origin`.